Code/Framework/AzFramework/AzFramework/Physics/RigidBodyBus.h

```diff
@@ -79,13 +81,66 @@ namespace Physics
 
     using RigidBodyRequestBus = AZ::EBus<RigidBodyRequests>;
 
-
+    //! Notifications interface for a rigid body (static or dynamic).
     class RigidBodyNotifications
         : public AZ::ComponentBus
     {
+    private:
+        template<class Bus>
+        struct RigidBodyNotificationsConnectionPolicy : public AZ::EBusConnectionPolicy<Bus>
+        {
+            static void Connect(
+                typename Bus::BusPtr& busPtr,
+                typename Bus::Context& context,
+                typename Bus::HandlerNode& handler,
+                typename Bus::Context::ConnectLockGuard& connectLock,
+                const typename Bus::BusIdType& id = 0)
+            {
+                AZ::EBusConnectionPolicy<Bus>::Connect(busPtr, context, handler, connectLock, id);
+
+                AZ::Entity* entity = nullptr;
+                AZ::ComponentApplicationBus::BroadcastResult(entity, &AZ::ComponentApplicationBus::Events::FindEntity, id);
+                if (entity)
+                {
+                    // Only immediately dispatch if the entity is already active, otherwise when
+                    // entity will get activated it will make the notifications itself.
+                    const AZ::Entity::State entityState = entity->GetState();
+                    if (entityState == AZ::Entity::State::Active)
+                    {
+                        // Only immediately dispatch if the entity is a RigidBodyRequestBus' handler.
+                        RigidBodyRequestBus::EnumerateHandlersId(
+                            id,
+                            [handler, id](const RigidBodyRequests* rigidBodyhandler)
```

Comment:
question: Do we definitely want to capture the `handler` by value here? Could this instead by a reference? (not sure if it's cheap to copy or not)

Reply:
I'll give it a go passing as reference.